airbyte-server/src/main/java/io/airbyte/server/handlers/SchedulerHandler.java

```diff
@@ -409,4 +434,19 @@ private JobInfoRead readJobFromResult(final ManualOperationResult manualOperatio
     return jobConverter.getJobInfoRead(job);
   }
 
+  private boolean containsBreakingChange(final CatalogDiff diff) {
+    AtomicBoolean isBreaking = new AtomicBoolean(false);
+    diff.getTransforms().stream().forEach(streamTransform -> {
+      if (streamTransform.getTransformType() != TransformTypeEnum.UPDATE_STREAM) {
+        return;
+      }
+      streamTransform.getUpdateStream().stream().forEach(fieldTransform -> {
```

Comment:
Nit: we should consider `takeWhile` here instead of `foreach` in order to be able to breakout of the loop as soon as possible. I wonder if it can have an impact on big catalogs.

Reply:
I'm not sure exactly how this would work. I think takeWhile continues to operate on each item until it reaches something that does not match the predicate, and then it will not operate on that field, so I'm not sure we could actually set isBreaking to true in that case..but maybe you're thinking about a different way of doing this?

Reply:
I just updated it to use `anyMatch` instead which is probably cleaner and I think does break out early